service/storage_service.cc

```diff
@@ -6926,6 +6926,16 @@ future<> storage_service::stream_tablet(locator::global_tablet_id tablet) {
         ranges_per_endpoint[leaving_replica_ip].emplace_back(range);
         streamer->add_rx_ranges(table.schema()->ks_name(), std::move(ranges_per_endpoint));
         co_await streamer->stream_async();
+
+        // If new pending tablet replica needs splitting, streaming waits for it to complete.
+        // That's to provide a guarantee that once migration is over, the coordinator can finalize
+        // splitting under the promise that compaction groups of tablets are all split, ready
+        // for the subsequent topology change.
```

Comment:
We could also migrate during splitting (and then the coordinator would only consider the destination for completion).

This solves the following scenario:

1. coordinator begin splitting a huge table
2. operator decommissions node
3. operator has to wait for the split to complete